clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java

```diff
@@ -241,13 +262,40 @@ public EhcacheEntityResponse invoke(ClientDescriptor clientDescriptor, EhcacheEn
                                                   " Check your server configuration and define at least one offheap resource.");
       }
 
+      //All the invokes are blocked till passive evicted keys are invalidated by server under lock
+      //Once invalidations are done, other threads will just move on by checking the status
+      if (!invalidationForPassiveEvictedKeysSent.get()) {
```

Comment:
Rather than ditching the logic, we can simply opt for sending invalidations without stopping other threads to make progress as we discussed last time. wdyt ?